symfit/core/argument.py

```diff
@@ -1,70 +1,36 @@
 
-    The ``Argument`` class also makes DRY possible in defining ``Argument``'s: it uses ``inspect`` to read the lhs of the
-    assignment and uses that as the name for the ``Argument`` is none is explicitly set.
+    If no name is explicitly provided a name will be generated.
 
     For example::
 
-        x = Variable()
-        print(x.name)
-        >> 'x'
+        y = Variable()
+        print(y.name)
+        >> 'x_0'
+
+        y = Variable('y')
+        print(y.name)
+        >> 'y'
     """
-    def __new__(cls, name=None, **assumptions):
+    def __new__(cls, name=None, *args, **assumptions):
         assumptions['real'] = True
-        # Super dirty way? to determine the variable name from the calling line.
-        if not name or not isinstance(name, basestring):
-            frame, filename, line_number, function_name, lines, index = inspect.stack()[1]
-
-            # Check if the script is running in an exe
-            # If this is the case, symfit cannot find the vaiable name from the calling line and it will search
-            # in the exe directory to find the source. Make sure to copy the required source files for this to work.
-            if getattr(sys, 'frozen', False):
-                basedir = sys._MEIPASS #  Find the exe base dir
-                fname = os.path.basename(filename)
-                # Find the source file in the pyinstaller directory
-                source_file = None
-                for root, dirs, files in os.walk(basedir):
-                    if fname in files:
-                        source_file = os.path.join(root, fname)
-                        break
-
-                if source_file is None:
-                    raise IOError("Source code file not found")
-
-                # Get the correct line from the source code
-                l = 0
-                with open(source_file, 'r') as fid:
-                    for line in fid:
-                        l+=1
-                        if l == line_number:
-                            lines = line
-            caller = lines[0].strip()
-            if '==' in caller:
-                pass
-            else:
-                try:
-                    terms = caller.split('=')
-                except ValueError:
-                    generated_name = name
-                else:
-                    generated_name = terms[0].strip()  # lhs
-                return super(Argument, cls).__new__(cls, generated_name, **assumptions)
-        return super(Argument,cls).__new__(cls, name, **assumptions)
+        # Generate a dummy name
+        if not name:
+            generated_name = '{}_{}'.format(cls._argument_name, cls._argument_index)
+            instance = super(Argument, cls).__new__(cls, generated_name, **assumptions)
+            instance._argument_index = cls._argument_index
+            cls._argument_index += 1
+            return instance
+        else:
+            return super(Argument, cls).__new__(cls, name, **assumptions)
```

Comment:
Isn't it better to let `super().__new__` (i.e. sympy) generate a name? Otherwise:
```
if not name:
    name = '{}_{}'.format(cls._argument_name, cls._argument_index)
    cls._argument_index += 1
instance = super(Argument, cls).__new__(cls, name, **assumptions)
instance._argument_index = cls._argument_index  # Do we need to do this? I think not.

```